src/AggregateFunctions/AggregateFunctionHistogram.cpp

```diff
@@ -76,16 +67,30 @@ class AggregateFunctionHistogramData
     Mean lower_bound;
     Mean upper_bound;
 
-    // Weighted values representation of histogram.
-    WeightedValue points[0];
+    // Weighted values representation of histogram - Structure of Arrays format.
+    Mean mean[0];
+    Weight weight[0];
```

Comment:
It's strange to have two zero-length arrays one after another. This means that they appear in the same place, and it is likely illegal.